piwarsimulator.py

```diff
@@ -166,50 +185,64 @@ def sigint_handler(signal_received, frame):
-    from controllers.GripperController import GripperController
-    from controllers.LauncherController import LauncherController
-    from sensors.DistanceSensor import DistanceSensor
-elif args.mode.lower() == "control_simulation":
-    # TODO fake control hardware, real sensors
-    util.import_serial()
-    from controllers.SimulatedMovementController import SimulatedMovementController
-    from controllers.SimulatedGripperController import SimulatedGripperController
+match args.mode:
+    case OperationMode.Simulation:
+        # fake control hardware, fake sensors
+        from controllers.SimulatedMovementController import SimulatedMovementController
+        from controllers.SimulatedGripperController import SimulatedGripperController
+        from sensors.SimulatedLineOfSight import SimulatedLineOfSight
+        from sensors.SimulatedBeamSensor import SimulatedBeamSensor
+
+        if not util.is_true(args.simplevision):
+            from sensors.SimulatedVision360 import SimulatedVision360 as SimulatedVision
+        else:
+            from sensors.SimulatedReducedVision import (
+                SimulatedReducedVision as SimulatedVision,
+            )
+    case OperationMode.SensorSimulation:
+        # real control hardware, fake sensors
+        util.import_serial()
+        from controllers.SimulatedMovementController import SimulatedMovementController
+        from controllers.SimulatedGripperController import SimulatedGripperController
+        from controllers.MovementController import MovementController
+        from controllers.GripperController import GripperController
+        from sensors.SimulatedLineOfSight import SimulatedLineOfSight
+        from sensors.SimulatedBeamSensor import SimulatedBeamSensor
+
+        if not util.is_true(args.simplevision):
+            from sensors.SimulatedVision360 import SimulatedVision360 as SimulatedVision
+        else:
+            from sensors.SimulatedReducedVision import (
+                SimulatedReducedVision as SimulatedVision,
+            )
+    case OperationMode.Control:
+        # real control hardware, real sensors
+        util.import_serial()
+        from controllers.MovementController import MovementController
+        from controllers.GripperController import GripperController
+        from controllers.LauncherController import LauncherController
+        from sensors.DistanceSensor import DistanceSensor
+        from sensors.Vision360 import Vision360
+
+    case OperationMode.ControlSimulation:
+        # TODO fake control hardware, real sensors
+        util.import_serial()
+        from controllers.SimulatedMovementController import SimulatedMovementController
+        from controllers.SimulatedGripperController import SimulatedGripperController
+        from sensors.DistanceSensor import DistanceSensor
+        # TODO: Probably needs beam sensor here
+        from sensors.Vision360 import Vision360
+
+    case OperationMode.EverythingSimButVision:
```

Comment:
I like this solution (having a separate mode for real vision only)

Reply:
Yeah - I hadn't quite realised what you were doing with that setup there, but I figured just adding a new thing in was the most sensible solution.